extension/parquet/CMakeLists.txt

```diff
@@ -26,6 +26,63 @@ set(PARQUET_EXTENSION_FILES
     zstd_file_system.cpp)
 
 if(NOT CLANG_TIDY)
+  # copy over checks from snappy
+  include(CheckCXXSourceCompiles)
+  check_cxx_source_compiles(
```

Comment:
Elsewhere in the code I can see that we use `__builtin_expect`, `__builtin_ctzll` and `__builtin_prefetch` when we have `#ifdef __GNUC__`. I think that's a pretty safe default - perhaps that works as well, or do we also want to use it when compiling with clang?

Reply:
Also to add - we don't use cmake when compiling the Python package, so this won't be present there in any case

Reply:
I think this is not very portable indeeded, I would use the builtins as we do elsewhere.

Fun facts, clang also defines __GNUC__, since they sort of use the same API (hand-waves), so that will work on both gcc and clang.